appinventor/components/src/com/google/appinventor/components/runtime/Spinner.java

```diff
@@ -190,16 +471,24 @@ public void AfterSelecting(String selection){
   public void onItemSelected(AdapterView<?> parent, View view, int position, long id){
     //prevent AfterPicking triggering when component has just been instantiated.
     if (!isInitialized) {
-      isInitialized = true;
-      return;
+        isInitialized = true;
```

Comment:
Line 400 in the Elements property setter causes the first user-initiated selection from calling AfterSelection:
     //avoid firing off the OnItemSelect when component is initialized or data is changed
     isInitialized = false;

This is broken badly and basically makes the Elements method unusable!

This is not a problem for ElementsFromString.